zkvm/src/txlog.rs

```diff
@@ -76,6 +84,66 @@ impl TxID {
         entry.commit_to_transcript(&mut t);
         t.challenge_bytes(b"merkle.leaf", result);
     }
+
+    /// Computes the Merkle proof of inclusion required to validate an
+    /// entry at index i, not including the original item i.
+    /// Fails when requested index is out of bounds
+    pub fn proof(
```

Comment:
Would be good to separate public API from internal recursive API (especially if it's going to be rewritten into a non-recursive one later). E.g. have `pub fn proof(list, index) -> Vec<MerkleHash>` and keep transcript creation and recursion behind it with private `fn subproof(t, list, index &mut vec)` (which may also become infailable, as the public API already has checked for the possible errors).

Reply:
See how `from_log` is public and `node` is recursive and private because it assumes good inputs that are provided internally-only.